init.lua

```diff
@@ -1186,10 +1200,38 @@ function PaperWM:moveWindow(window, frame)
     end)
 end
 
+---add or remove focused window from the floating layer and retile the space
+function PaperWM:toggleFloating()
+    local window = Window.focusedWindow()
+    if not window then
+        self.logger.d("focused window not found")
+        return
+    end
+
+    local id = window:id()
+    local space = nil
+    is_floating[id] = not is_floating[id]
+    local persisted = {}
+    for k, v in pairs(is_floating) do
+        if v then table.insert(persisted, k) end
+    end
+    hs.settings.set(IsFloatingKey, persisted)
+
+    if is_floating[id] then
+        space = index_table[id]['space']
+        self:removeWindow(window, true)
```

Comment:
`removeWindow()` and `addWindow()` will return the space of the removed/added window

Reply:
Oh I completely missed that—thanks!